packages/common/src/contract/validator.ts

```diff
@@ -0,0 +1,86 @@
+import { ContractDeploymentInfo } from './interface'
+
+export const validateCellDeps = (info: ContractDeploymentInfo) => {
+  if (info.cellDeps) {
+    if (!Array.isArray(info.cellDeps)) {
+      throw new Error(`cellDeps must be an array`)
+    }
+
+    if (
+      !info.cellDeps.every((cellDep) => {
+        if ('name' in cellDep) {
+          return true
+        } else if (!('cellDep' in cellDep)) {
+          false
+        } else {
+          if (
+            'outPoint' in cellDep.cellDep &&
+            'txHash' in cellDep.cellDep.outPoint &&
+            'index' in cellDep.cellDep.outPoint &&
+            ['code', 'depGroup'].includes(cellDep.cellDep.depType)
+          ) {
+            return true
+          } else {
+            return false
+          }
+        }
+      })
+    ) {
+      throw new Error(`cellDeps must be an array of { name: string } or { cellDep: CellDep }`)
+    }
+  }
+
+  return info
+}
```

Comment:
Nested too much.

1. `!Array.isArray(info.cellDeps)` can be at the top
2. the condition of `if` at line 10 is complex and should be a function

Reply:
> Nested too much.
> 
> 1. `!Array.isArray(info.cellDeps)` can be at the top
> 2. the condition of `if` at line 10 is complex and should be a function

This validator only validates the `cellDeps` only when the contract has dependent cells. So if the `info.cellDeps` is undefined, the function will return directly.